### PR TITLE
Fix #4 - add spiders for treasury expenditure and receipts

### LINKED ISSUE #4: Create treasury expenditure and receipts crawlers

### PR DESCRIPTION
**Pull request by curioswati**

### CODE REVIEW

File: scraper/scraper/spiders/ddo_code_collector.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+class DDOCodeCollector(scrapy.Spider):
+    '''
+    Collect DDO Codes for treasuries.
+    '''
+    name = 'ddo_collector'
+    start_urls = ['https://himkosh.nic.in/eHPOLTIS/PublicReports/wfrmDDOAllocationExpenditure.aspx']
+
+    def parse(self, response):
+        # create form request that will automatically collect form data from the page response.
+        fr = scrapy.FormRequest.from_response(response, callback=self.collect_ddo_code)
```
Comment (gggodhwani) [inline scraper/scraper/spiders/ddo_code_collector.py]: Let's try verbose names "fr"

File: scraper/scraper/spiders/ddo_code_collector.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                             meta={'treasury': treasury_code})
+
+    def collect_ddo_code(self, response):  # pylint:disable=no-self-use
+        '''
+        create a CSV file with DDO names and DDO Codes.
+        '''
+        ddo_selector = response.xpath('id("ddlDDOCode")/option')
+
+        treasury_code = response.meta.get('treasury')
+        filepath = os.path.join(DATASET_PATH, '{}_ddo_codes.csv'.format(treasury_code))
```
Comment (gggodhwani) [inline scraper/scraper/spiders/ddo_code_collector.py:49]: If possible add a date based version control. Swati & Gaurav decided to go forward with Github itself to ensure version controlling, we will generate a token based pull request and someone needs to manually review it.

**Review by gggodhwani — APPROVED** [review]
Looks good to me!

File: scraper/scraper/spiders/ddo_code_collector.py (hunk 2 above)
Comment (curioswati) [reply scraper/scraper/spiders/ddo_code_collector.py:49]: will be taken forward with #8

File: scraper/scraper/spiders/ddo_code_collector.py (hunk 1 above)
Comment (curioswati) [reply scraper/scraper/spiders/ddo_code_collector.py]: resolved in https://github.com/CivicDataLab/hp-fiscal-data-explorer/pull/5/commits/9745cf28ddcf179c5a5dc28ab31cc683a2cd09b8

**Pull request merged**